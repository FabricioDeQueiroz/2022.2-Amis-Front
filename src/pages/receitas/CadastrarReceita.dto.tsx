export interface ReceitasCadastrarDTO {
  nome: string;
  descricao: string;
  ingredientes: [string];
  modo_preparo: [string];
}
