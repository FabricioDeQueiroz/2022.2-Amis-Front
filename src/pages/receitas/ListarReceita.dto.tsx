export interface ListarReceitaDTO {
  nome: string;
  descricao: string;
  ingredientes: [string];
  modo_preparo: [string];
}
