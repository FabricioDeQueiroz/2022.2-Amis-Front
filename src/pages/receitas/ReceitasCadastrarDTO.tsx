export interface ReceitasCadastrarDTO {
  nome: string;
  ingredientes: string;
  modo_preparo: string;
}
