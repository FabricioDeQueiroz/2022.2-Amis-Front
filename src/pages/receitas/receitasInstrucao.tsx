import React from "react";
import styled from "styled-components";
import { Navbar } from "../../shared/components/Navbar/navbar";
import Title from "../../shared/components/Title/Title";
import { Footer } from "../../shared/components/Footer/footer";
import PrimaryButton from "../../shared/components/PrimaryButton/PrimaryButton";

const DivPresentation = styled.div`
  background-color: ${(props) => props.theme.colors.gray};
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 30px;
`;

const DivPresentationInner = styled.div`
  background-color: white;
  display: flex;
  flex-direction: column;
  width: 100%;
  border-radius: 25px;
  padding: 30px;
`;

const DivIngredientes = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 40px;
`;

const DivButtons = styled.div`
  width: 100%;
  display: inline-flex;
  justify-content: flex-end;
  gap: 20px;
  margin: 0 auto;
  padding-top: 0px;
`;

export function ReceitasInstrucao() {
  return (
    <div>
      <Navbar />
      <DivPresentation>
        <DivPresentationInner>
          <Title fontSize={40} fontWeight={500}>
            RECEITA 1 {/* {receita?.nome} // Esse vai ser o fixo */}
          </Title>
          <DivButtons>
            <PrimaryButton
              text={"Excluir"}
              handleClick={() => console.log("EXCLUIR")}
            />
            <PrimaryButton
              text={"Editar receita"}
              handleClick={() => console.log("EDITAR RECEITA")}
            />
          </DivButtons>
          <DivIngredientes>
            <Title fontSize={35} fontWeight={400}>
              INGREDIENTES
            </Title>
          </DivIngredientes>
        </DivPresentationInner>
      </DivPresentation>
      <Footer />
    </div>
  );
}
