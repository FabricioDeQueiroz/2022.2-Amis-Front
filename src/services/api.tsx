/* eslint-disable @typescript-eslint/restrict-template-expressions */
/* eslint-disable no-template-curly-in-string */
import axios from "axios";
import { getUserLocalStorage } from "./auth";

const userLocalStorage = getUserLocalStorage();

const api = axios.create({
  baseURL: "https://service-amis.azurewebsites.net/",
  headers: {
    Authorization: `Bearer ${userLocalStorage?.token}`,
  },
});

export default api;
