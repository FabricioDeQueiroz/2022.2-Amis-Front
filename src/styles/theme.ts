const theme = {
  colors: {
    primary: "#da4d3d",
    white: "#FFF",
    black: "#000",
  },
};

export default theme;
