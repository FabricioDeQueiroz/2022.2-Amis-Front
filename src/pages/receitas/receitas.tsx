import React, { useState } from "react";
import styled from "styled-components";
import { Navbar } from "../../shared/components/Navbar/navbar";
import PrimaryButton from "../../shared/components/PrimaryButton/PrimaryButton";
import Title from "../../shared/components/Title/Title";
import { Footer } from "../../shared/components/Footer/footer";
import {
  Box,
  Card,
  CardActionArea,
  CardContent,
  Modal,
  TextField,
  Typography,
} from "@mui/material";
import { useForm } from "react-hook-form";
import { Form } from "react-router-dom";
import { ReceitasCadastrarDTO } from "./ReceitasCadastrarDTO";
import axios from "axios";
import { toast } from "react-toastify";

const style = {
  position: "absolute",
  top: "50%",
  left: "50%",
  transform: "translate(-50%, -50%)",
  width: 600,
  bgcolor: "background.paper",
  border: "none",
  boxShadow: 24,
  p: 4,
  padding: "50px",
  height: "85%",
  overflow: "hidden",
  overflowY: "scroll",
};

const FormText = styled.h1`
  color: #525252;
  font-size: 18px;
  font-weight: 400;
  text-align: left;
  padding-bottom: 25px;
`;

const DivPresentation = styled.div`
  background-color: ${(props) => props.theme.colors.gray};
  display: flex;
  flex-direction: column;
  padding: 45px 120px 120px 100px;
`;

const DivHeaderReceitas = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 30px;
`;

const DivCards = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-content: space-between;
  justify-content: space-between;
  align-items: center;
`;

const Container = styled.div`
  width: 100%;
`;

export function Receitas() {
  const [openCad, setOpenCad] = useState(false);
  const handleOpen = () => setOpenCad(true);
  const handleClose = () => setOpenCad(false);

  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm({});

  const registerReceitas = async (data: any) => {
    const receita = {
      nome: data.nome,
      ingredientes: data.ingredientes,
      modo_preparo: data.modo_preparo,
    } as ReceitasCadastrarDTO;

    await axios
      .post("https://service-amis.azurewebsites.net/receitas/", receita)
      .then((response) => {
        console.log(response.status);
        toast.success("Receita criada com sucesso!");
      })
      .catch((err) => console.warn(err));
  };

  return (
    <Container>
      <Navbar />
      <DivPresentation>
        <DivHeaderReceitas>
          <Title fontSize={40} fontWeight={600}>
            RECEITAS
          </Title>
          <PrimaryButton text={"Cadastrar receita"} handleClick={handleOpen} />
        </DivHeaderReceitas>
        <Modal open={openCad} onClose={handleClose}>
          <Box sx={style}>
            <FormText
              style={{
                textAlign: "center",
                fontWeight: "bold",
                fontSize: 30,
              }}
            >
              Cadastrar uma nova receita
            </FormText>
            <Form onSubmit={handleSubmit(registerReceitas)}>
              <TextField
                id="outlined-nome_receita"
                label="Nome"
                required={true}
                {...register("nome")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />
              <TextField
                id="outlined-ingredientes"
                label="Ingredientes"
                required={true}
                {...register("ingredientes")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />
              <TextField
                id="outlined-modo_preparo"
                label="Modo de Preparo"
                required={true}
                {...register("modo_preparo")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />
              <PrimaryButton text={"Cadastrar receita"} />
            </Form>
          </Box>
        </Modal>
        <DivCards>
          <Card sx={{ maxWidth: 330, borderRadius: 7, padding: 2, margin: 2 }}>
            <CardActionArea>
              <CardContent>
                <Typography gutterBottom variant="h5" component="div">
                  Biscoito ou Bolacha
                </Typography>
                <Typography variant="body2" color="text.secondary">
                  Lorem ipsum dolor sit amet, consectetur adipisicing elit.
                  Quaerat laboriosam, asperiores fugiat quae ad corporis.
                </Typography>
              </CardContent>
            </CardActionArea>
          </Card>
        </DivCards>
      </DivPresentation>
      <Footer />
    </Container>
  );
}
