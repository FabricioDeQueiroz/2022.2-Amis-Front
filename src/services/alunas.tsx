import { AlunasCadastrarDTO } from "../pages/alunas/dtos/AlunasCadastrarDTO";
import api from "./api";

export const cadastraAluna = async (payload: AlunasCadastrarDTO) => {
  return await api.post("/alunas/", payload).then((response: any) => response);
};

export const listaAlunas = async () => {
  return await api.get("/alunas/").then((response: any) => response);
};

export const apagaAluna = async (alunaId: string) => {
  return await api
    .delete("/alunas/" + alunaId)
    .then((response: any) => response);
};

export const editaAluna = async (
  alunaId: string,
  alunaEdit: AlunasCadastrarDTO
) => {
  return await api
    .put("/alunas/" + alunaId, alunaEdit)
    .then((response: any) => response);
};