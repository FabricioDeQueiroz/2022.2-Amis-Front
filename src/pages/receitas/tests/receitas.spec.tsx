import { Receitas } from "../receitas";
import {
  getByAltText,
  getByLabelText,
  getByPlaceholderText,
  getByRole,
  getByTestId,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React, { Component } from "react";
import { BrowserRouter, BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import theme from "../../../styles/theme";
import renderer from "react-test-renderer";
import { QueryClientProvider } from "react-query";
import { queryClient } from "../../../services/queryClient";

//describe("receitas", () => {
//it("Receitas", async () => {
    // eslint-disable-next-line react/react-in-jsx-scope
//  const { getByText, queryByText, getByLabelText, findByText, debug } =
//    render(
        // eslint-disable-next-line react/react-in-jsx-scope
//      <QueryClientProvider client={queryClient}>
//        <Router>
//          <ThemeProvider theme={theme}>
//            <Receitas />
//          </ThemeProvider>
//        </Router>
//      </QueryClientProvider>
//    );

//  const inputNome = getByLabelText("Nome");
//  const inputIngrediente = getByLabelText("Ingrediente");

//  void userEvent.type(inputNome, "PÃ£o de queijo");
//  void userEvent.type(inputIngrediente, "Farinha");

//  const addButtonCadastrar = getByText("Cadastrar Receita");
//  void userEvent.click(addButtonCadastrar);

//  await waitFor(() => {
//  const inputModoPreparo = getByLabelText("Modo de preparo");
//    void userEvent.type(inputModoPreparo, "Assar por 30 minutos");
//  });

//  await waitFor(() => {
//    const addButtonCadastrar = getByText("Cadastrar");
//    void userEvent.click(addButtonCadastrar);
//    expect(getByLabelText("Modo de preparo")).toBeInTheDocument();
//    });
//  });
//});


describe("Snapshot", () => {
  it("Deve corresponder ao Snapshot", () => {
    const tree = renderer
      .create(
        <QueryClientProvider client={queryClient}>
          <BrowserRouter>
            <ThemeProvider theme={theme}>
              <Receitas />
            </ThemeProvider>
          </BrowserRouter>
        </QueryClientProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
