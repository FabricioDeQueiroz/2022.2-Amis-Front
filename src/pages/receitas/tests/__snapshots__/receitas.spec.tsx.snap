// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Deve corresponder ao Snapshot 1`] = `
<div
  className="sc-ksBlkl fhBUpU"
>
  <div
    className="sc-gswNZR kZkAEs"
  >
    <div
      className="sc-dkrFOg bktbFR"
    >
      <div
        className="sc-hLBbgP PqNIR"
      >
        <h1
          className="sc-eDvSVe kinkDz"
        >
          AMIS
        </h1>
        <a
          href="/"
          onClick={[Function]}
        >
          <button
            className="sc-jSUZER jcgnhb"
            onClick={[Function]}
          >
            Home
          </button>
        </a>
        <a
          href="/receitas"
          onClick={[Function]}
        >
          <button
            className="sc-jSUZER dJarrl"
            onClick={[Function]}
          >
            Receitas
          </button>
        </a>
      </div>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="sc-bcXHqe hosLZH"
          type="submit"
        >
          Área Logada
        </button>
      </a>
    </div>
  </div>
  <div
    className="sc-ipEyDJ jKoUzE"
  >
    <div
      className="sc-csuSiG eVWyhP"
    >
      <h1
        className="sc-gKPRtg cQNkqa"
        fontSize={40}
        fontWeight={600}
      >
        RECEITAS
      </h1>
    </div>
    <div
      className="sc-eDWCr jZlDS"
    />
  </div>
  <div
    className="sc-iBYQkv dWoHUu"
  >
    <a
      href="https://www.instagram.com/amismulherescriativas/"
    >
      <img
        className="sc-ftTHYK cNAVlE"
        src="test-file-stub"
        style={
          {
            "marginBottom": "20px",
            "marginRight": "20x",
            "width": "40px",
          }
        }
      />
    </a>
    <a
      href="https://pt-br.facebook.com/InstitutoMulheresCriativas/"
    >
      <img
        className="sc-ftTHYK cNAVlE"
        src="test-file-stub"
        style={
          {
            "marginBottom": "20px",
            "marginRight": "20px",
            "width": "20px",
          }
        }
      />
    </a>
    <div>
      <span
        className="sc-pyfCe cmKoG"
      >
        © 2022. All rights reserved.
      </span>
    </div>
  </div>
</div>
`;
